spec/models/manageiq/providers/redhat/infra_manager/operations/snapshot_spec.rb

```diff
@@ -9,7 +9,7 @@
       allow(@closeable_snapshots_service).to receive(:snapshot_service)
         .with(snapshot.uid_ems) { @snapshot_service }
       allow(vm).to receive(:closeable_snapshots_service).with(any_args).and_return(@closeable_snapshots_service)
-      allow(Snapshot).to receive(:find_by_id).with(snapshot.id).and_return(snapshot)
+      allow(Snapshot).to receive(:find_by).with(:id => snapshot.id).and_return(snapshot)
```

Comment:
@chessbyte The reason this doesn't work as intended is because while the implementation of `find_by_*` directs to the class (`Snapshot`), `find_by` is a finder method found on `ActiveRecord::Relation` (returned by `snapshots`). 

This is usually not such a weird issue because a scope is mocked or an actual Snapshot object is used (a real db lookup).

To do the least amount of changes to make the specs pass as written, however, you can mock the relation itself doing the following instead:

```ruby
allow(vm).to receive(:snapshots).and_return(double(:find_by => snapshot))
```